### ISSUE TITLE
Parsing error when setting a blacklist in config.json

### ISSUE THREAD

**Issue by mazito** [near/stakewars-iii#83]
When I add a `blacklist` array to `config.json` and restart the node, I get a parsing error:

![image](https://user-images.githubusercontent.com/1116909/182914110-569ef77d-d23b-4e43-8cb6-50aa3df9ac7b.png)

~~~
"blacklist": [
"ed25519:33o4JQC8hjJdMgyWG94cBsfqQhtpLmLRx45qV8u7ggiQ@38.242.159.111:24567","ed25519:53piE91PJT2qhRegMFZ3sUAoVnokxuwdvc8F91KNcwdp@51.140.201.208:24567","ed25519:8tJpyE3ZaxVVPMNwjPKrjaFX9eXSDuEgKFVCwQhGftLB@185.207.250.23:24567","ed25519:D89MYJ1gQD1eYizdPbmzUcwo2wYXZgCn2bpe3M4BEKB1@38.242.159.113:24567","ed25519:6cZTF3m96vNpG8sXkDg3wfK5hvB1UBpYuYSH847SnDGx@185.229.119.250:24567","ed25519:CfYC9iUmcaEs7J7VZ2FCX5Q5D3WtkfTDizHHpoqiXJiG@86.48.2.165:24567","ed25519:6gnvwhWhsCFu73QXcfKMxcJSQf9vSW7QXCbE7zXKSuVt@157.245.82.130:24567","ed25519:2J74P3ZFFrNpwm1qbHcAK8BJD7EPqAbu4hEJdaFUuJZt@167.235.21.149:24567","ed25519:GtYU4BVDPfU1FoHZSbCdcr25tmnkKWE9KFaAubnJGjnx@188.34.178.187:24567","ed25519:HMPS6RDwsNj3KXKJYs9bpA7yB4KGFCnXYxVYgTJb3M4m@143.198.15.59:24567","ed25519:AUrsUPtWMjPq8qCvKcyamh4XQPBc25JYTkRofaxBQeXr@86.48.2.74:24567","ed25519:6pJudpwEFX6Q6jhTPbS8wzf2WRsvtnR3MLa3Zfi6UgsP@161.97.136.124:24567","ed25519:9RUJrkYjzB6BvtsCKK43sBzouyRwjqqiqajYb5U8p37Y@194.5.152.123:24567","ed25519:5DsKHTXjzhnz6GjNgYqKHSPCGStxqZbfCzDh2yED1MLz@137.184.61.129:24567","ed25519:9TscLJ4k8YYj8Y12a5ZeiF1aHD4wQeKnGMLYW4Es89o4@38.242.247.7:24567","ed25519:2SY4Xpb6srFdy1E5QQyZiziz6Na9FhkrV7SHzLkxE3cd@149.28.232.200:24567","ed25519:FLuvkRta4g5oG5XnxUzhs4VTM1bdqSFkThHQYt7AtAcU@20.77.69.243:24567"
],
~~~

Using a string also gives a parsing error:
~~~
"blacklist": 
"ed25519:33o4JQC8hjJdMgyWG94cBsfqQhtpLmLRx45qV8u7ggiQ@38.242.159.111:24567,ed25519:53piE91PJT2qhRegMFZ3sUAoVnokxuwdvc8F91KNcwdp@51.140.201.208:24567"
,
~~~

Question:
 - What is the right format for the blacklist ?
 - Is there a limit to the number of nodes being blacklisted ?

Thanks !

**Comment by joesixpack** [comment]
Try it without the port?  Without the IP?

**Comment by stiavnik** [comment]
Tried various formats of **blacklist**: nothing worked, panic on AddrParseError(Socket) as described above.

**Whitelist**: at least neard runs, not sure if it's an improvement though. At the moment it's running for about half hour and I'm still "syncing":true (lost sync during various tests and can't get back).

Steps (command based on mazito from issue 84):
`curl -s -d '{"jsonrpc": "2.0", "method": "network_info", "id": "dontcare", "params": [null]}' -H 'Content-Type: application/json' 127.0.0.1:3030`
then manually parse all 16 records containing IP adresses into config.json (attached _00config-shardnet-whitelist.txt_).

Logs (journalctl -u neard) timestamps (attached _log.zip_):
Aug 06 19:39:29 - starting with default config.json from wget.
Aug 06 20:53:48 - starting with whitelist added (16 IPs).
Aug 06 21:08:56 - 3 boot nodes also added to whitelists (19 IPs total).

Running zilinear2.factory.shardnet.near on:
  "build": "1.1.0-2580-g78ef2f558",
  "rustc_version": "1.62.1"
(yes this was "release" only shortly but after jumping back and forth without visible changes I am on this version at the moment)
[00config-shardnet-whitelist.txt](https://github.com/near/stakewars-iii/files/9275792/00config-shardnet-whitelist.txt)
[log.zip](https://github.com/near/stakewars-iii/files/9275793/log.zip)

Not sure if this will help something but at least I tried.
